### PR TITLE
Define pr, master and loop targets.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
The master loop currently fails because it tries to upload snapshots. To ease the definitions of different targets we separate build and upload logic.

### CODE REVIEW

File: ci/pipeline [hunk 1]
```diff
@@ -75,13 +75,14 @@ def uploadTarballToS3(): Option[awsClient.Artifact] = utils.stage("Upload Packag
...
+ * @param snapshot Indicates whether we want to upload the snapshot.
  * @return Version and artifact description of Marathon build.
  */
-def createAndUploadPackages(): (String, Option[awsClient.Artifact]) = {
+def createAndUploadPackages(snapshot: Boolean = true): (String, Option[awsClient.Artifact]) = {
   val version = createPackages()
   val docsPath = PACKAGE_DOCS_DIR / s"marathon-docs-$version.tgz"
-  val artifact = uploadTarballToS3()
-  awsClient.upload(docsPath)
+  val artifact = if(snapshot) uploadTarballToS3() else None
```
Comment (alenkacz) [inline ci/pipeline]: shouldn't this condition be inverted? The title says we do NOT want to upload when snapshot build but this says if snapshot, then upload... or am I missing something?

**Comment by jeschkies** [comment]
Do not merge yet! The archive does not work yet.

File: ci/pipeline (hunk 1 above)
Comment (jeschkies) [reply ci/pipeline]: Good point. The logic is as I want it but the name `snapshot` is not fitting 🤔.

File: Jenkinsfile [hunk 2]
```diff
@@ -13,16 +13,18 @@ ansiColor('gnome-terminal') {
           sh """sudo -E ci/pipeline jenkins"""
         }
       } finally {
-        junit(allowEmptyResults: true, testResults: 'target/test-reports/*.xml')
-        junit allowEmptyResults: true, testResults: 'target/test-reports/*integration/*.xml'
+        //junit(allowEmptyResults: true, testResults: 'target/test-reports/*.xml')
```
Comment (zen-dog) [inline Jenkinsfile]: is this temporary?

Comment (jeschkies) [reply Jenkinsfile]: Of course :). This is for faster iterations on Jenkins.

File: ci/pipeline [hunk 3]
```diff
@@ -30,12 +30,13 @@ def compileAndTest(logFileName: String): Unit = utils.stage("Compile and Test")
 
   def run(cmd: String *) = utils.runWithTimeout(30.minutes, logFileName)(cmd)
 
-  run("sbt", "clean", "test", "integration:test", "scapegoat")
+  //run("sbt", "clean", "test", "integration:test", "scapegoat")
```
Comment (zen-dog) [inline ci/pipeline]: don't forget to un-comment

Comment (jeschkies) [reply ci/pipeline]: Thanks, Mr. Obvious ^^.

Comment (jeschkies) [reply ci/pipeline]: You are right though. I've added the `wip` tag.

File: ci/pipeline [hunk 4]
```diff
@@ -75,13 +76,14 @@ def uploadTarballToS3(): Option[awsClient.Artifact] = utils.stage("Upload Packag
 /**
  * Packages Marathon and uploads its artifacts alongside sha1 checksum to S3.
  *
+ * @param upload Indicates whether we want to upload the tarball package.
  * @return Version and artifact description of Marathon build.
  */
-def createAndUploadPackages(): (String, Option[awsClient.Artifact]) = {
+def createAndUploadPackages(upload: Boolean = true): (String, Option[awsClient.Artifact]) = {
```
Comment (zen-dog) [inline ci/pipeline]: This grew and got out of hand. We should refactor here a little. Lets do 2 methods:
- `createTarballPackage()`
- `uploadArtifatct(artifact: Artifact)`

File: ci/pipeline [hunk 5]
```diff
@@ -75,13 +76,14 @@ def uploadTarballToS3(): Option[awsClient.Artifact] = utils.stage("Upload Packag
...
   val version = createPackages()
   val docsPath = PACKAGE_DOCS_DIR / s"marathon-docs-$version.tgz"
-  val artifact = uploadTarballToS3()
-  awsClient.upload(docsPath)
+  val artifact = if(upload) uploadTarballToS3() else None
+  if(upload) awsClient.upload(docsPath)
 
   createDockerAndLinux()
```
Comment (zen-dog) [inline ci/pipeline]: May be we should split that in `createDockerImage` and `createLinuxPackages`

File: ci/pipeline (hunk 4 above)
Comment (zen-dog) [reply ci/pipeline]: And we should put all creation of packages (Tarball, Docker, Linux) into one stage.

File: ci/pipeline [hunk 6]
```diff
@@ -199,14 +202,14 @@ def run(publish: Boolean = false): (String, Option[awsClient.Artifact]) = {
 @main
 def jenkins(): Unit = {
   if(utils.isPullRequest) {
```
Comment (zen-dog) [inline ci/pipeline]: What do you think about calling 2 distinct `@main` methods from here where each one of those should then call it's own targets? E.g. PR's don't publish artifacts and master builds do etc.

Comment (kensipe) [reply ci/pipeline]: we are likely to add another distinct `@main` as well... I'm expecting release to be it's own path.

**Comment by zen-dog** [comment]
We just discussed this with Karsten and I'm ready to lower the acceptance bar 😄 But there will be a few minor improvements that Karsten will push soon

**Comment by zen-dog** [comment]
Karsten's proposing that we should leverage `sbt` more extensively, effectively making most of this script into sbt's own targets.

File: ci/pipeline (hunk 6 above)
Comment (jeschkies) [reply ci/pipeline]: Done.

**Pull request merged**